### ISSUE TITLE
Remove strong dependency between modals and their sources

### ISSUE THREAD

**Issue by andrewtavis** [activist-org/activist#657]
### Terms

- [X] I have searched [open and closed issues](https://github.com/activist-org/activist/issues?q=is%3Aissue)
- [X] I agree to follow activist's [Code of Conduct](https://github.com/activist-org/activist/blob/main/.github/CODE_OF_CONDUCT.md)

### Issue

As of now [ModalBase](https://github.com/activist-org/activist/blob/main/frontend/components/modal/ModalBase.vue) is creating a strong dependency between the modal and the normal that's triggering it. We've discussed this most recently in the last developer sync as well as in conversations afterwards. It would be good if this were removed and there was a bit more independence with the modal being triggered by a prop that's then more customizable.

@rigojr brought this up and has expressed interest in working on it :)

**Comment by andrewtavis** [comment]
Hey @rigojr 👋 Went ahead and made the issue here so we don't forget it 😊 Let me know if you still have interest in working on it!

**Comment by rigojr** [comment]
Hi @andrewtavis I would love to make this refactor.

Let me know when I could start, I understand that someone is doing some work over there, so I could start after that, let me know what you think.

**Comment by andrewtavis** [comment]
Thanks for the willingness to help, @rigojr! Let's wait for a PR for the share modal component to get merged and then we can start :)

**Comment by andrewtavis** [comment]
Hey @rigojr 😊 Let me know if a call on this would be helpful at some point! We also now have a modal for displaying the organization application, and the share modal is also live that would ideally link to the QR code modal, but this is broken because of this...

Hope you're well! :)

**Comment by andrewtavis** [comment]
Hey @rigojr 👋 Checking in with you on this, as this issue's now much more important given that we're using modals in a lot of other places on the platform. Would you have time to take a look at this, or we could also do a call where we discuss the work to be done so I can implement?

**Comment by andrewtavis** [comment]
Changes to fix this are coming along shortly :) Noting some things about the changes here 😇

The following code is within `ModalBase` and all child components in the `modal` directory:

```ts
const props = defineProps<{
  isOpen: boolean;
}>();

const modalIsOpen = computed(() => props.isOpen);
const modalShouldClose = ref(false);

const emit = defineEmits(["closeModal"]);
const closeModal = () => {
  modalShouldClose.value = true;
  emit("closeModal");
  modalShouldClose.value = false;
};
```

Modal components that use `ModalBase` for their styling include the following:

```vue
<ModalBase
  @closeModal="handleCloseModal"
  :isOpen="modalShouldClose == false ? modalIsOpen : false"
>
...
</ModalBase>
```

And the instances of the individual modals have the following:

```vue
<ElementThatOpensModal
  @click="openModal()"
  @keydown.enter="openModal()"
/>
<ModalComponent
  @closeModal="handleCloseModal"
  :isOpen="modalIsOpen"
/>
```

and

```ts
const modalIsOpen = ref(false);

function openModal() {
  modalIsOpen.value = true;
}

const handleCloseModal = () => {
  modalIsOpen.value = false;
};
```

It's all a bit more verbose then I'd ideally like it to be, but what the above entails is a passage of `modalIsOpen` to the modal instance that then is transferred from its `props.isOpen` to its own `modalIsOpen` via `const modalIsOpen = computed(() => props.isOpen);`, with the same process happening in `ModalBase` to open the modal. 

Closing the modal required a little trick :) We have the `ref` `modalShouldClose` that's set to `false`, with the value being switched to `true` in calls of `closeModal` within `ModalBase` and other modals that closes the modal and includes an emit of `closeModal` back up the chain all the way to the original component. After the emit, `modalShouldClose` is switched back to `false` such that the modal can be reopened again via `:isOpen="modalShouldClose == false ? modalIsOpen : false"`.

What a process, but the modals are separate and we're now able to open modals within modals. The big thing is that we can only have one modal within any given component, but then this can be worked around in architectural decisions 😊

**Comment by andrewtavis** [comment]
Closed via f7ba8bd 🥳 The one thing that's a bit weird now is that we can open modals over modals, but I've made #845 to look into solutions where opening one modal would close ones that are open :)

Thanks for bringing this to our attention, @rigojr! With modals being such an important part of how the platform will work, it's been so important to get your feedback on them. Hope all's well and that we'll see you soon!
